app/mailers/payment_mailer.rb

```diff
@@ -16,8 +16,7 @@ def authorize_payment(payment)
   end
 
   def authorization_required(payment)
-    @payment = payment
-    @order = @payment.order
+    @order = payment.order
```

Comment:
Now looking at the views, I found that they were only referring to the order, not the payment. So instead of making `order` a local variable, I removed `@payment` as variable and changed the views to access `@order` directly.